presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java

```diff
@@ -106,9 +114,18 @@ private FileSystem getNonCachingFileSystem()
         return environment.getFileSystem(context, cacheStoragePath);
     }
 
-    private RubixConfigurationInitializer initializeRubix()
+    private void initializeCachingFileSystem(RubixConfig rubixConfig)
             throws IOException
     {
+        cachingFileSystem = getCachingFileSystem(initializeRubix(rubixConfig));
+        cachingInitialized = true;
```

Comment:
It does not work correctly. Note that you can get exception from `getCachingFileSystem`, after `initializeRubix` is already called. 
Then the `closeRubix` will not be called and cleanup will not be made. Not sure if that is big of a deal. But if we want the cleanup to happen we need to guard on individual fields which are closed/cleaned up, not on whole routine.

Reply:
You are right. I will add explicit null checks